### PR TITLE
Layer manager

### PR DESCRIPTION
**Pull request by MarkShleifer**

### CODE REVIEW

**Comment by MarkShleifer** [comment]
Need to do unit tests

File: src/app/packages/menu-items/layers-manager/event-args/node-activation-changed-event-args.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { TreeNode } from 'angular-tree-component';
+
+export class NodeActivationChangedEventArgs {
+    constructor(public node: TreeNode, public newState: boolean) {
```
Comment (eldadcohen1) [inline src/app/packages/menu-items/layers-manager/event-args/node-activation-changed-event-args.ts]: Please define properties explicitly.
public nod : TreeNode;
constructor(node : TreeNode)

File: src/app/packages/menu-items/layers-manager/index.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export { LayersManagerModule } from './layers-manager.module';
```
Comment (eldadcohen1) [inline src/app/packages/menu-items/layers-manager/index.ts]: Add new line for eof.

File: src/app/packages/menu-items/layers-manager/layer-tree/layer-tree.component.html [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    <div id="node{{ node.data.id }}">
+      <input [indeterminate]="isNodeIndeterminate(node)" type="checkbox" [checked]="node.isActive">
+      <span [class]="node.data.className" [class.title]="true">{{ node.data.name }}</span>
+    </div>
+  </template>
+  <template #loadingTemplate let-node>
+    <div [class]="node.data.className + 'Loading'">Loading {{ node.data.title }}...</div>
+  </template>
+</tree-root>
+<br/>
```
Comment (eldadcohen1) [inline src/app/packages/menu-items/layers-manager/layer-tree/layer-tree.component.html]: Add EOF new line

File: src/app/packages/menu-items/layers-manager/layer-tree/layer-tree.component.ts [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+import { NodeActivationChangedEventArgs } from '../event-args/node-activation-changed-event-args';
+import { TreeActionMappingServiceService } from '../services/tree-action-mapping-service.service';
+import { TreeNode, TreeComponent } from 'angular-tree-component';
+import { ILayerTreeNode } from '../models/layer-tree-node';
+
+@Component({
+  selector: 'app-layer-tree',
+  templateUrl: './layer-tree.component.html',
+  styleUrls: ['./layer-tree.component.scss'],
+  providers: [TreeActionMappingServiceService]
```
Comment (eldadcohen1) [inline src/app/packages/menu-items/layers-manager/layer-tree/layer-tree.component.ts]: Fixed the name ServiceService

File: src/app/packages/menu-items/layers-manager/layer-tree/layer-tree.component.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+})
+
+export class LayerTreeComponent implements OnInit, AfterViewInit{
+
+  private options;
+
+  @ViewChild(TreeComponent) treeComponent: TreeComponent;
+
+  @Input() public source: any[];
```
Comment (eldadcohen1) [inline src/app/packages/menu-items/layers-manager/layer-tree/layer-tree.component.ts]: The source should be an interface for ILayerTreeNode

File: src/app/packages/menu-items/layers-manager/layers-manager/layers-manager.component.html [hunk 6]
```diff
@@ -0,0 +1,3 @@
+<div class="layer-manager">
+    <app-layer-tree [source]=nodes></app-layer-tree>
+</div>
```
Comment (eldadcohen1) [inline src/app/packages/menu-items/layers-manager/layers-manager/layers-manager.component.html]: Add EOF new line

File: src/app/packages/menu-items/layers-manager/layers-manager/layers-manager.component.scss [hunk 7]
```diff
@@ -0,0 +1,4 @@
+div.layer-manager {
+    width: 400px;
+    margin: 0px 0px 20px 0px;
+}
```
Comment (eldadcohen1) [inline src/app/packages/menu-items/layers-manager/layers-manager/layers-manager.component.scss]: Add EOF new line

File: src/app/packages/menu-items/layers-manager/layers-manager/layers-manager.component.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+  styleUrls: ['layers-manager.component.scss']
+})
+export class LayersManagerComponent implements OnInit {
+
+  private nodes: ILayerTreeNode[];
+  constructor() { }
+
+  ngOnInit() {
+
+    let rootNode: ILayerTreeNode = {
```
Comment (eldadcohen1) [inline src/app/packages/menu-items/layers-manager/layers-manager/layers-manager.component.ts]: This is all an hard coded mock data.
Please integrate this with mock service and not with hardcoded mock.
Add a develop configuration and producrtion configuration.

File: src/app/packages/menu-items/layers-manager/services/tree-action-mapping-service.service.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@angular/core';
+import { TREE_ACTIONS, KEYS, IActionMapping } from 'angular-tree-component';
+
+@Injectable()
+export class TreeActionMappingServiceService {
```
Comment (eldadcohen1) [inline src/app/packages/menu-items/layers-manager/services/tree-action-mapping-service.service.ts]: Fixed the name to Service and not for ServiceService

File: package.json [hunk 10]
```diff
@@ -25,7 +25,6 @@
     "angular-tree-component": "^3.2.3",
     "@ngrx/core": "^1.2.0",
     "@ngrx/store": "^2.2.1",
-    "@ng-bootstrap/ng-bootstrap": "^1.0.0-alpha.22",
```
Comment (eldadcohen1) [inline package.json]: good

File: src/app/packages/menu-items/layers-manager/event-args/node-activation-changed-event-args.ts [hunk 11]
```diff
@@ -1,7 +1,11 @@
 import { TreeNode } from 'angular-tree-component';
 
 export class NodeActivationChangedEventArgs {
-    constructor(public node: TreeNode, public newState: boolean) {
+    public node: TreeNode;
+    public newState: boolean;
 
+    constructor(node: TreeNode, newState: boolean) {
+        this.node = node;
+        this.newState = newState;
```
Comment (eldadcohen1) [inline src/app/packages/menu-items/layers-manager/event-args/node-activation-changed-event-args.ts]: Great 👑

File: src/app/packages/menu-items/layers-manager/layer-tree/layer-tree.component.spec.ts [hunk 12]
```diff
@@ -3,11 +3,11 @@ import { DebugElement } from '@angular/core';
 import { By } from '@angular/platform-browser';
 import { LayerTreeComponent } from './layer-tree.component';
 import { LayersManagerModule } from '../layers-manager.module';
-import { CoreModule } from '../../../core/core.module';
-import { ILayerTreeNode } from '../models/layer-tree-node'
+import { CoreModule } from '@ansyn/core/core.module';
+import { ILayerTreeNode } from '@ansyn/core';
```
Comment (eldadcohen1) [inline src/app/packages/menu-items/layers-manager/layer-tree/layer-tree.component.spec.ts]: greate 🖐 👌

**Pull request closed without merging**